### PR TITLE
Send poster statistics weekly via email

### PR DESCRIPTION
**Pull request by ibarrae**

### CODE REVIEW

File: api/posters/statistics/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  </table>
+  `
+}
+
+export type RawPosterData = Pick<PosterDocument, 'posterData'>
+
+type PerLanguageStats = {[k in string]: number}
+
+export const sendEmail = async (content: string) => {
+  const to = (process.env.SEND_POSTER_ANALYTICS_RECIPIENTS || '').split(',')
```
Comment (JoseRFelix) [inline api/posters/statistics/helpers.ts:90]: If Recipients are not defined, shouldn't we stop the execution? What do you think?

File: api/posters/statistics/send.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      now,
+    )
+
+    const totalPosters = posterDocuments.length
+    const emailContent = getEmailContent(posterDocuments, totalPosters)
+
+    await sendEmail(emailContent)
+  } catch (e) {
+    logger.error(`Error sending posters analytics`, e)
+  }
```
Comment (JoseRFelix) [inline api/posters/statistics/send.ts:32]: We should reject the promise for observability. We can use something like `return Promise.reject(createHttpError(500, 'ERROR sending poster statistics'))`

File: serverless.yml [hunk 3]
```diff
@@ -210,6 +210,15 @@ functions:
...
           cors: true
 
+  sendPosterStatistics:
+    handler: posters/statistics/send.sendPosterStatistics
+    environment:
+      SEND_POSTER_ANALYTICS_RECIPIENTS:
+        ${ssm:/${self:custom.stage}/ses/poster_analytics_recipients,
+        env:SEND_POSTER_ANALYTICS_RECIPIENTS}
+    events:
+      - schedule: cron(0 0 * * 1) # Notify at the start of the week
```
Comment (JoseRFelix) [inline serverless.yml:220]: :+1:

File: api/posters/statistics/send.spec.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        expect.stringContaining('<td>typescript</td><td>1</td>'),
+      )
+
+      expect(sendEmailSpy).toHaveBeenCalledWith(
+        expect.stringContaining('<td>haskell</td><td>1</td>'),
+      )
+
+      expect(sendEmailSpy).not.toHaveBeenCalledWith(
+        expect.stringContaining('<td>golang</td><td>1</td>'),
+      )
```
Comment (JoseRFelix) [inline api/posters/statistics/send.spec.ts:95]: Maybe also adding a `toMatchInlineSnapshot()` could help to make sure the email shell is consistent.

File: api/posters/statistics/send.spec.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      const posters = await PosterModel.scan().exec()
+      posters.forEach(async poster => {
+        await poster.delete()
+      })
+      const sendEmailSpy = jest
+        .spyOn(helpers, 'sendEmail')
+        .mockResolvedValueOnce()
+      await sendPosterStatistics()
+      expect(sendEmailSpy).toHaveBeenCalledWith(
+        expect.stringContaining('No new posters this week'),
```
Comment (JoseRFelix) [inline api/posters/statistics/send.spec.ts:34]: We can also use `toMatchInlineSnapshots` here to make sure the email is consistent.

**Review by JoseRFelix — APPROVED** [review]
Thanks, looks good! Left some non-blocking comments. Let me know what you think.

File: api/posters/statistics/helpers.ts (hunk 1 above)
Comment (ibarrae) [reply api/posters/statistics/helpers.ts:90]: Sure, that sounds like a good idea! Thanks for the suggestion :+1:

File: api/posters/statistics/send.ts (hunk 2 above)
Comment (ibarrae) [reply api/posters/statistics/send.ts:32]: Great suggestion!

**Pull request merged**